### ISSUE TITLE
Include auto-generated content without the source page (How to use on_files?)

### ISSUE THREAD

**Issue by shodgson11** [mkdocs/mkdocs#1647]
I want to include some auto-generated pages inside of other pages without showing the auto-generated page itself.  Right now I'm using the automated nav so I don't have to maintain the list of pages (but that means any extra md files in the docs dir get added to the nav).

I added a folder at the same level as the docs folder (to hide the auto-generated files from the navigation).  I have successfully used markdown_include.include to insert the auto-generated content from this folder into a live page using relative paths (`{!../auto-generated/include-me.md!}`).  This works fine with mkdocs serve, but the mkdocs build ignores the content from this auto-generated folder location.

I've seen several references for using `on_files` to include files outside of the docs dir at build time, but I'm having trouble finding a concrete example of how to use it.  How can I use `on_files` to include files from the auto-generated folder without putting them into the auto-generated nav?

\- project
-- auto-generated
-- docs
-- site
-- mkdocs.yml

**Comment by shodgson11** [comment]
Thanks for the reply!

I don't remember seeing any misleading docs, just reporting the behavior I'm observing where `mkdocs serve` allows files from the `auto-generated` folder to be included in the live preview while `mkdocs build` does not (by default) include my content from the `auto-generated` folder in the build output.

#714 seems like it would allow me to work around my issue here, though given the choice I would prefer that the include extension be allowed to read from files outside the default collection without putting the source pages in the nav.
Also, as much as I don't want to manually maintain the master nav list, doing so would resolve my issue.

If there's no other magic bullet out there for the selective inclusion, feel free to close this and I'll watch the other ticket.

**Comment by waylan** [comment]
> just reporting the behavior I'm observing where mkdocs serve allows files from the auto-generated folder to be included in the live preview while mkdocs build does not (by default) include my content from the auto-generated folder in the build output.

There should be no difference between `build` and `serve`. If there is, then that is a bug. we would need a minimal replaceable example to bug it though.

Regarding the nav, it occurs to me that you might be able to save a list of the files included within your plugin and then in a nav event, walk the nav and remove those items from the nav. I realize that is a less-than-ideal solution, but it would provide a working solution today.

**Comment by shodgson11** [comment]
Ok, I found the source of my problem.  I had removed the markdown_include base_path from mkdocs.yml while already in serve mode and this did not negatively affect the page I was viewing (even with a browser refresh).  However, restarting serve or running build with that line missing results in the include not working because it can no longer find the file to be included.

On the bright side, the attached project does exactly what I was looking for - include a snippet from a folder outside of docs where the source file is not included in the auto nav.
[include-proj.zip](https://github.com/mkdocs/mkdocs/files/2450641/include-proj.zip)

Thanks!
